docs/references-docs-in-vfs.md

```diff
@@ -0,0 +1,301 @@
+such cases. The goal of this document is to offer a way to do that, and it is
+called _References_.
+
+The references of a file are listed in its JSON-API representation in the
+`references` field, within the `relationships` object of `data`.
+
+### Example
+
+```json
+{
+  "data": {
+    "type": "io.cozy.files",
+    "id": "9152d568-7e7c-11e6-a377-37cbfb190b4b",
+    "meta": {
+      "rev": "1-0e6d5b72",
+    },
+    "attributes": {
+      "type": "file",
+      "name": "hello.txt",
+      "md5sum": "86fb269d190d2c85f6e0468ceca42a20",
+      "created_at": "2016-09-19T12:38:04Z",
+      "updated_at": "2016-09-19T12:38:04Z",
+      "tags": [],
+      "size": 12,
+      "executable": false,
+      "class": "document",
+      "mime": "text/plain"
+    },
+    "relationships": {
+      "parent": {
+        "links": {
+          "related": "/files/fce1a6c0-dfc5-11e5-8d1a-1f854d4aaf81"
+        },
+        "data": {
+          "type": "io.cozy.files",
+          "id": "fce1a6c0-dfc5-11e5-8d1a-1f854d4aaf81"
+        }
+      }
+    },
+    "links": {
+      "self": "/files/9152d568-7e7c-11e6-a377-37cbfb190b4b"
+    }
+  }
+}
+```
+
+## Routes
+
+### PATCH /files/:file-id/relationships/references
+
+Replace the references for a file with the new ones.
+
+#### Request
+
+```http
+PATCH /files/9152d568-7e7c-11e6-a377-37cbfb190b4b/relationships/references HTTP/1.1
+Content-Type: application/vnd.api+json
+Accept: application/vnd.api+json
+```
+
+```json
+{
+  "data": [
+    { "type": "io.cozy.playlists", "id": "94375086-e2e2-11e6-81b9-5bc0b9dd4aa4" }
+  ]
+}
+```
+
+#### Response
+
+```http
+HTTP/1.1 204 No Content
+Content-Type: application/vnd.api+json
+```
+
+### POST /files/:file-id/relationships/references
+
+Add on a file one or more references to documents
+
+#### Request
+
+```http
+POST /files/9152d568-7e7c-11e6-a377-37cbfb190b4b/relationships/references HTTP/1.1
+Content-Type: application/vnd.api+json
+Accept: application/vnd.api+json
+```
+
+```json
+{
+  "data": [
+    { "type": "io.cozy.playlists", "id": "f2625cc0-e2d6-11e6-a0d5-cfbbfb141af0" }
+  ]
+}
+```
+
+#### Response
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/vnd.api+json
+```
+
+```json
+{
+  "data": [
+    { "type": "io.cozy.playlists", "id": "94375086-e2e2-11e6-81b9-5bc0b9dd4aa4" },
+    { "type": "io.cozy.playlists", "id": "f2625cc0-e2d6-11e6-a0d5-cfbbfb141af0" }
+  ]
+}
+```
+
+### DELETE /files/:file-id/relationships/references
+
+Remove one or more references to documents on a file
+
+#### Request
+
+```http
+DELETE /files/9152d568-7e7c-11e6-a377-37cbfb190b4b/relationships/references HTTP/1.1
+Content-Type: application/vnd.api+json
+Accept: application/vnd.api+json
+```
+
+```json
+{
+  "data": [
+    { "type": "io.cozy.playlists", "id": "f2625cc0-e2d6-11e6-a0d5-cfbbfb141af0" }
+  ]
+}
+```
+
+#### Response
+
+```http
+HTTP/1.1 204 No Content
+Content-Type: application/vnd.api+json
+```
+
+### GET /files/:type/:doc-id/relationships/references
```

Comment:
[nitpick] While it makes sense in echo groups structure to have these routes as child of /files/, it seems weird from an user/rest POV. `/data/:type/:doc-id/relationships/reference` ?